cmd/rpc.go

```diff
@@ -175,3 +191,77 @@ func startNativeSpaceBridgeRpcServer(ctx context.Context, wg *sync.WaitGroup) {
 	server := rpc.MustNewNativeSpaceBridgeServer(&config)
 	go server.MustServeGraceful(ctx, wg, config.Endpoint, rpcutil.ProtocolHttp)
 }
+
+// startDebugSpaceRpcServer starts RPC server for geth debug API
+func startDebugSpaceRpcServer(ctx context.Context, wg *sync.WaitGroup) {
+	httpEndpoint := viper.GetString("debugrpc.endpoint")
+	ethNodeRPCURL := viper.GetString("node.router.ethnoderpcurl")
+	logrus.Debug("Debug Space RPC server HTTP endpoint=", httpEndpoint, ", ethNodeRPCURL=", ethNodeRPCURL)
+	client, err := gethrpc.DialHTTP(ethNodeRPCURL)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to create rpc client")
+	}
+	debugRouter = node.NewNodeRpcRouter(client)
+
+	http.Handle("/", &ForwardHandler{})
+	http.ListenAndServe(httpEndpoint, nil)
+}
+
+var debugRouter *node.NodeRpcRouter
+
+type ForwardHandler struct{}
+
+func (h *ForwardHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	ip := whitelist.GetIPFromRequestEnv(req)
+	logrus.Debug("remote IP: ", ip)
+	valid, err := whitelist.IsIPValid(ip)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if valid == false {
+		body, err := ioutil.ReadAll(req.Body)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		var data map[string]interface{}
+		err = json.Unmarshal(body, &data)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		ip, _ := data["ip"]
```

Comment:
Why does the JSON payload have an "ip" field?

Reply:
it's id, fixed.

Reply:
btw, it seems that i have missed one commit here, will fix it.